src/workerd/io/compatibility-date-test.c++

```diff
@@ -180,6 +180,118 @@ KJ_TEST("compatibility flag parsing") {
   // Can explicitly disable flag that's enabled for all dates.s
   expectCompileCompatibilityFlags("2021-05-17", {"r2_internal_beta_bindings"}, "()", {},
       CompatibilityDateValidation::FUTURE_FOR_TEST, true, false);
+
+  // nodejs_compat implies nodejs_compat_v2 on or after 2024-08-05
+  expectCompileCompatibilityFlags("2024-08-05", {"nodejs_compat"},
+      "(formDataParserSupportsFiles = true,"
+      " fetchRefusesUnknownProtocols = true,"
+      " esiIncludeIsVoidTag = false,"
+      " obsolete3 = false,"
+      " durableObjectFetchRequiresSchemeAuthority = true,"
+      " streamsByobReaderDetachesBuffer = true,"
+      " streamsJavaScriptControllers = true,"
+      " jsgPropertyOnPrototypeTemplate = true,"
+      " minimalSubrequests = true,"
+      " noCotsOnExternalFetch = true,"
+      " specCompliantUrl = true,"
+      " globalNavigator = true,"
+      " captureThrowsAsRejections = true,"
+      " r2PublicBetaApi = true,"
+      " obsolete14 = false,"
+      " noSubstituteNull = true,"
+      " transformStreamJavaScriptControllers = true,"
+      " r2ListHonorIncludeFields = true,"
+      " exportCommonJsDefaultNamespace = true,"
+      " obsolete19 = false,"
+      " webSocketCompression = true,"
+      " nodeJsCompat = true,"
+      " obsolete22 = false,"
+      " specCompliantResponseRedirect = true,"
+      " workerdExperimental = false,"
+      " durableObjectGetExisting = false,"
+      " httpHeadersGetSetCookie = true,"
+      " dispatchExceptionTunneling = true,"
+      " serviceBindingExtraHandlers = false,"
+      " noCfBotManagementDefault = true,"
+      " urlSearchParamsDeleteHasValueArg = true,"
+      " strictCompression = true,"
+      " brotliContentEncoding = true,"
+      " strictCrypto = true,"
+      " rttiApi = false,"
+      " webgpu = false,"
+      " cryptoPreservePublicExponent = true,"
+      " vectorizeQueryMetadataOptional = true,"
+      " unsafeModule = false,"
+      " jsRpc = false,"
+      " noImportScripts = true,"
+      " nodeJsAls = false,"
```

Comment:
shouldn't `nodeJsAls` be an implied flag that is activated automatically when `nodeJsCompat` is on?

Reply:
No, it's a separate flag intended to make `AsyncLocalStorage` available regardless of the rest of `nodejs_compat`. The `nodeJsCompat`/v2 flags already enable it. There's no need for it to be implied.